hack/run-e2e.sh

```diff
@@ -103,50 +77,20 @@ e2e_args=(
     --clean-start=true
     --delete-namespace-on-failure=false
     # tidb-operator e2e flags
-    # TODO make these configurable via environments
     --operator-tag=e2e
     --operator-image=${TIDB_OPERATOR_IMAGE}
+    --e2e-image=${E2E_IMAGE}
     --test-apiserver-image=${TEST_APISERVER_IMAGE}
     --tidb-versions=v3.0.2,v3.0.3,v3.0.4,v3.0.5
     --chart-dir=/charts
     -v=4
     ${@:-}
 )
-# We don't attach into the container because the connection may lost.
-# Instead we print logs and check the result repeatedly after the pod is Ready.
-$KUBECTL_BIN ${kubectl_args[@]:-} -n ${NS} run tidb-operator-e2e --generator=run-pod/v1 --image $E2E_IMAGE \
-    --env NAMESPACE=$NS \
-    --labels app=webhook \
-    --serviceaccount=tidb-operator-e2e \
-    --image-pull-policy=IfNotPresent \
-    --restart=Never \
-    --command -- ${e2e_args[@]}
-echo "info: wait for e2e pod to be ready"
-ret=0
-$KUBECTL_BIN ${kubectl_args[@]:-} -n ${NS} wait --for=condition=Ready pod/tidb-operator-e2e || ret=$?
-if [ $ret -ne 0 ]; then
-    echo "error: failed to wait for the e2e pod to be ready, printing its logs"
-    $KUBECTL_BIN ${kubectl_args[@]:-} -n ${NS} logs tidb-operator-e2e
-    exit 1
-fi
 
-echo "info: start to print e2e logs and check the result"
-execType=0
-while true; do
-    if [[ $execType -eq 0 ]]; then
-        $KUBECTL_BIN ${kubectl_args[@]:-} -n ${NS} logs -f tidb-operator-e2e || true
-    else
-        $KUBECTL_BIN ${kubectl_args[@]:-} -n ${NS} logs --tail 1 -f tidb-operator-e2e || true
-    fi
-    phase=$($KUBECTL_BIN ${kubectl_args[@]:-} -n ${NS} get pods tidb-operator-e2e -ojsonpath='{.status.phase}')
-    if [[ "$phase" == "Succeeded" ]]; then
-        echo "info: e2e succeeded"
-        exit 0
-    elif [[ "$phase" == "Failed" ]]; then
-        echo "error: e2e failed, phase: $phase"
-        exit 1
-    fi
-    # if we failed on "$KUBECTL_BIN ${kubectl_args[@]:-} logs -f", try to print one line each time
-    # TODO find a better way to print logs starting from the last place
-    execType=1
-done
+docker run --rm \
+    --net=host \
+    -v $KUBECONFIG:/etc/kubernetes/admin.conf \
```

Comment:
It's better to mount this as readonly.

Reply:
done in https://github.com/pingcap/tidb-operator/pull/1304/commits/6e094d012de80360d2754d8084ca0ecaeb99bc6d